src/implementations/vanilla/components/tabs/tabs.js

```diff
@@ -334,6 +374,100 @@ export class Tabs {
     this.moreButton.setAttribute('aria-expanded', false);
     this.dropdown.classList.remove('ecl-tabs__dropdown--show');
   }
+
+  /**
+   * Tabs onKeydown handler.
+   * @param {Event} e
+   */
+  onKeydown(e) {
+    const tgt = e.currentTarget;
+    let flag = false;
+
+    switch (e.key) {
+      case 'ArrowLeft':
+        this.moveFocusToPreviousTab(tgt);
+        flag = true;
+        break;
+
+      case 'ArrowRight':
+        this.moveFocusToNextTab(tgt);
+        flag = true;
+        break;
+
+      case 'Home':
+        this.moveFocusToTab(this.firstTab);
+        flag = true;
+        break;
+
+      case 'End':
+        this.moveFocusToTab(this.lastTab);
+        flag = true;
+        break;
+
+      default:
+        break;
+    }
+
+    if (flag) {
+      e.stopPropagation();
+      e.preventDefault();
+    }
+  }
+
+  /**
+   * @param {HTMLElement} currentTab tab element
+   */
+  moveFocusToTab(currentTab) {
+    if (currentTab.closest('.ecl-tabs__dropdown')) {
+      this.dropdown.classList.add('ecl-tabs__dropdown--show');
+    } else {
+      this.dropdown.classList.remove('ecl-tabs__dropdown--show');
+    }
+    currentTab.focus();
+    return this;
+  }
+
+  /**
+   * @param {HTMLElement} currentTab tab element
+   */
+  moveFocusToPreviousTab(currentTab) {
+    const index = this.tabsKey.indexOf(currentTab);
+
+    if (this.isMobile) {
+      if (currentTab !== this.firstTab) {
+        this.moveFocusToTab(this.tabsKey[index - 1]);
+        this.shiftTabs('prev');
+      }
+      return;
+    }
+
+    if (currentTab === this.firstTab) {
+      this.moveFocusToTab(this.lastTab);
+    } else {
+      this.moveFocusToTab(this.tabsKey[index - 1]);
+    }
+  }
+
+  /**
+   * @param {HTMLElement} currentTab tab element
+   */
+  moveFocusToNextTab(currentTab) {
+    const index = this.tabsKey.indexOf(currentTab);
+
+    if (this.isMobile) {
+      if (currentTab !== this.lastTab) {
+        this.moveFocusToTab(this.tabsKey[index + 1]);
+        this.shiftTabs('next');
+      }
+      return;
+    }
+
+    if (currentTab === this.lastTab) {
+      this.moveFocusToTab(this.firstTab);
+    } else {
+      this.moveFocusToTab(this.tabsKey[index + 1]);
+    }
+  }
```

Comment:
i don't see the new listeners being destroyed together with the others in the destroy method..

Reply:
done